lightning/src/ln/outbound_payment.rs

```diff
@@ -245,7 +269,8 @@ impl PendingOutboundPayment {
 					session_privs.insert(session_priv)
 				},
 			PendingOutboundPayment::AwaitingInvoice { .. } |
-				PendingOutboundPayment::InvoiceReceived { .. } => { debug_assert!(false); false },
+				PendingOutboundPayment::InvoiceReceived { .. } |
+				PendingOutboundPayment::StaticInvoiceReceived { .. } => { debug_assert!(false); false },
```

Comment:
Oof. It looks like we'll need to change it so that if we try to `insert` a session priv into a payment that's in one of these states on startup, we'll immediately convert it into a `Retryable`. Tracked this on #3207 to be addressed there if that sounds good to you, since this PR is pretty big already.